src/components/GroupsPageComponents/components/GroupsList/GroupsList.tsx

```diff
@@ -16,21 +17,39 @@ export default function GroupsList({ groups, onSelectGroup }: GroupsListProps) {
     { field: 'teacher', header: 'Преподаватель' },
   ];
 
+  const { getStudentsCount, getTeacherName } = useGroupList();
+
   return (
-    <DataTable
-      value={groups}
-      className={styles.table}
-      onRowClick={({ index }) => onSelectGroup && onSelectGroup(groups[index])}
-      rowHover>
-      {columns.map(({ field, header }, index) => (
-        <Column
-          field={field}
-          header={header}
-          headerClassName={styles.header}
-          bodyClassName={styles.cell}
-          key={index}
-        />
-      ))}
-    </DataTable>
+    <>
+      <DataTable
+        value={groups}
+        className={styles.table}
+        onRowClick={({ index }) =>
+          onSelectGroup && onSelectGroup(groups[index])
+        }
+        rowHover>
+        {columns.map(({ field, header }, index) => (
+          <Column
+            field={field}
+            header={header}
+            headerClassName={styles.header}
+            bodyClassName={styles.cell}
+            key={index}
+            body={(rowData: Group) => {
+              switch (field) {
+                case 'studentsCount':
+                  return getStudentsCount(rowData);
+                case 'teacher':
+                  return getTeacherName(rowData);
+                case 'name':
+                  return rowData.name;
+                case 'status':
+                  return rowData.status;
+              }
```

Comment:
Давненько я такой штуки не видел
![image](https://github.com/user-attachments/assets/14ca8ca3-9adc-43e8-a140-589d9319047d)

Reply:
Заместо switch лучше вот так писать (если есть возможность, то заместо any тоже типизацию настроить): 
```
type ExtractorKey = 'studentsCount' | 'teacher' | 'name' | 'status';

  const rowExtractors: Record<ExtractorKey,  any> = {
    studentsCount: rowData => getStudentCount(rowdata),
    teacher: rowData => getTeacherName(rowdata),
    name: rowData => rowdata.name,
    status: rowData => rowdata.status,
  };
```

а в просы уже вот так передавать:
```suggestion
            body={(rowData: Group) =>  {
             const extractor = rowExtractors[field]
             return extractor? extractor(rowData): '' 
            }```